src/components/navigation-control.js

```diff
@@ -40,7 +44,9 @@ export default class NavigationControl extends Component {
   _updateViewport(opts) {
     const {viewport} = this.context;
     const mapState = new MapState(Object.assign({}, viewport, opts));
-    this.props.onChangeViewport(mapState.getViewportProps());
+    // TODO(deprecate): remove this check when `onChangeViewport` gets deprecated
+    const onViewportChange = this.props.onChangeViewport || this.props.onViewportChange;
```

Comment:
To further expand, this is code that should be eventually removed from the codebase. Making it ugly now would prioritize its removal, I think.

I'm happy to use a util as well though if you feel strongly about doing it that way.